mapflow/functional/layer_utils.py

```diff
@@ -503,38 +504,50 @@ def download_results_callback(self, response: QNetworkReply, processing) -> None
         """
         self.dlg.processingsTable.setEnabled(True)
         # Avoid overwriting existing files by adding (n) to their names
-        output_path = os.path.join(self.dlg.outputDirectory.text(), processing.id_)
-        extension = '.gpkg'
-        if os.path.exists(output_path + extension):
+        output_path = Path(self.dlg.outputDirectory.text(), processing.id_).with_suffix(".gpkg")
+        if output_path.exists():
             count = 1
-            while os.path.exists(output_path + f'({count})' + extension):
+            while output_path.with_stem(processing.id_ + f"_{count}").exists():
                 count += 1
-            output_path += f'({count})' + extension
-        else:
-            output_path += extension
+            output_path = output_path.with_stem(processing.id_ + f"_{count}")
         transform = self.project.transformContext()
         # Layer creation options for QGIS 3.10.3+
         write_options = QgsVectorFileWriter.SaveVectorOptions()
         write_options.layerOptions = ['fid=id']
-        with open(os.path.join(self.temp_dir, os.urandom(32).hex()), mode='wb+') as f:
```

Comment:
Why would we even need self.temp_dir, if we are saving to a local file... Problem solved(?)